iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -244,102 +254,96 @@ Viewer.prototype._addIdToNode = function(node, id) {
 //                         Indicates type of element being wrapped
 //
 // All ixbrl-elements have "ivid" data added, which is a list of the ID
-// attribute(s) of corresponding IX element(s).  i.e. facts have fact IDs,
-// continuations have continuation IDs, footnotes have footnote IDs.  It's
-// possible for it to be a mix of different types.
+// attribute(s) of corresponding IX item(s).  Continuations have the IDs of
+// their head items (fact or footnotes).
+// "ivid" can be a mix of different types.
 //
 // Viewer._ixNodeMap is a map of these IDs to IXNode objects
 //
 // Viewer._docOrderIDIndex is a list of fact and footnote IDs in document
```

Comment:
I think it's now a list of {id,document} tuples, right?  (I don't JavaScript much, but I wonder if it would be clearer with an explicit DocumentScopedId class or something.)

Reply:
I've refactored the index itself into a separate class, which I think makes it more readable overall.  I've previously been reluctant to break out new classes because the pre-ES6 class syntax was so clunky, but it's probably about time we switched to using `class`.